Sources/MapboxNavigation/PreviewViewController.swift

```diff
@@ -0,0 +1,656 @@
+        setupFloatingButtons()
+        setupTopBannerContainerView()
+        setupBottomBannerContainerView()
+        setupConstraints()
+        setupStyleManager()
+    }
+    
+    open override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // Apply style each time `PreviewViewController` appears on screen
+        // (e.g. after active navigation).
+        styleManager.currentStyle?.apply()
+        setupPassiveLocationManager()
+        setupNavigationViewportDataSource()
+        subscribeForNotifications()
+        setupGestureRecognizers()
+    }
+    
+    open override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        setupOrnaments()
+        setupBottomBannerContainerViewLayoutConstraints()
+    }
+    
+    open override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        unsubscribeFromNotifications()
+        resetGestureRecognizers()
+    }
+    
+    open override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        if previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            setupBottomBannerContainerViewLayoutConstraints()
+        }
+    }
+    
+    open override func viewSafeAreaInsetsDidChange() {
+        super.viewSafeAreaInsetsDidChange()
+        
+        setupTopBannerContainerViewLayoutConstraints()
+    }
+    
+    // MARK: - UIViewController setting-up methods
+    
+    func setupNavigationView() -> NavigationView {
+        let frame = parent?.view.bounds ?? UIScreen.main.bounds
+        let navigationView = NavigationView(frame: frame)
+        navigationView.navigationMapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        navigationView.navigationMapView.delegate = self
+        // TODO: Move final destination annotation placement logic to `MapView` or `NavigationMapView`.
+        navigationView.navigationMapView.mapView.mapboxMap.onNext(event: .styleLoaded) { [weak self] _ in
+            guard let self = self else { return }
+            self.pointAnnotationManager = self.navigationView.navigationMapView.mapView.annotations.makePointAnnotationManager()
+            
+            if let finalDestinationAnnotation = self.finalDestinationAnnotation,
+               let pointAnnotationManager = self.pointAnnotationManager {
+                pointAnnotationManager.annotations = [finalDestinationAnnotation]
+                
+                self.finalDestinationAnnotation = nil
+            }
+        }
+        
+        return navigationView
+    }
+    
+    func setupBackButton() {
+        let backButton = BackButton(type: .system)
+        backButton.translatesAutoresizingMaskIntoConstraints = false
+        // TODO: Add localization.
+        backButton.setTitle("Back", for: .normal)
```

Comment:
While we’re at it, we should make the button title customizable. On iOS, a UINavigationController back button often inherits the name of the previous view controller on the stack when provided. I don’t know if there’s a straightforward way to do that automatically – maybe by crawling up the responder chain? But in that case, the button would need to be constrained to a maximum width and truncated, in case the previous view controller’s title is long.